config/v1/types_infrastructure.go

```diff
@@ -110,6 +115,53 @@ const (
 	IBMCloudProviderTypeVPC IBMCloudProviderType = "VPC"
 )
 
+// PlatformSpec holds the desired state specific to the underlying infrastructure provider
+// of the current cluster. Since these are used at spec-level for the underlying cluster, it
+// is supposed that only one of the spec structs is set.
+type PlatformSpec struct {
```

Comment:
indicate how these are vetted, validated, and promoted.

Reply:
> these

wdym by `these`? individual platform specs or something else.